app/src/main/java/com/example/criminalintent/presentation/dialogs/DatePickerFragment.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.criminalintent.presentation.dialogs
+
+import android.app.DatePickerDialog
+import android.app.Dialog
+import android.os.Bundle
+import android.widget.DatePicker
+import androidx.fragment.app.DialogFragment
+import java.util.*
+
+class DatePickerFragment : DialogFragment() {
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dateListener =
+            DatePickerDialog.OnDateSetListener { _: DatePicker, year: Int, month: Int, day: Int ->
+                val resultDate: Date = GregorianCalendar(year, month, day).time
+                val result = Bundle().apply {
+                    putSerializable(RESULT_DATE_KEY, resultDate)
+                }
+                val resultRequestCode = requireArguments().getString(ARG_REQUEST_CODE, "")
+                parentFragmentManager.setFragmentResult(resultRequestCode, result)
```

Comment:
This crashes when the fragment is not added to an activity.
>Throws:
IllegalStateException – if not associated with a transaction or host.

See `isAdded`